File: algorithms/easy/largest_local_values_in_a_matrix.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
      const int n = grid.size() - 2;
      vector<vector<int>> ans(n, vector<int>(n));
      for (auto i = 0; i < n; i++) {
        for (auto j = 0; j < n; j++) {
          for (auto di = 0; di < 3; di++) {
            for (auto dj = 0; dj < 3; dj++)
              ans[i][j] = max(ans[i][j], ans[i+di][j+dj]);
          }
        }
      }
      return ans;
    }
};

File: algorithms/easy/largest_local_values_in_a_matrix.rs
impl Solution {
    pub fn largest_local(grid: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let n = grid.len() - 2;
        let mut ans = vec![vec![0; n]; n];
        for i in 0..n {
            for j in 0..n {
                for di in 0..3 {
                    for dj in 0..3 {
                        ans[i][j] = ans[i][j].max(grid[i + di][j + dj]);
                    }
                }
            }
        }

        ans
    }
}
